packages/core/src/parser/magic-anchor-parser.ts

```
@@ -0,0 +1,180 @@
+      }
+    };
+  }
+  
+  // :A: api extract markers and prose from payload string
+  private static parsePayload(payload: string): { markers: string[]; prose?: string } {
+    // :A: ctx use regex to match marker patterns, then everything else is prose
+    // :A: ctx markers: word1, word2, word3 followed by space and prose
+    const markerPattern = /^([a-zA-Z0-9_@-]+(?:\([^)]*\))?(?:,\s*[a-zA-Z0-9_@-]+(?:\([^)]*\))?)*)\s+(.+)$/;
+    const match = payload.match(markerPattern);
+    
+    if (match && match[1]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)